torchtune/datasets/_chat.py

```diff
@@ -50,19 +50,21 @@ class ChatDataset(Dataset):
             (https://huggingface.co/docs/datasets/en/package_reference/loading_methods#datasets.load_dataset.path)
         convert_to_messages (Callable[[Mapping[str, Any]], List[Message]]): function that keys into the desired field in the sample
             and converts to a list of `Messages` that follows the llama format with the expected keys
-        chat_format (ChatFormat): template used to format the chat. If the placeholder variable
+        chat_format (Optional[ChatFormat]): template used to format the chat. If the placeholder variable
```

Comment:
let me know if this is too much details to add
```suggestion
        chat_format (Optional[ChatFormat]): template used to format the system, user, and assistant messages in the chat. This is used to add structured text around the actual messages, such as the [INST] tags in LLaMA2 and in Mistral. The extra text will still get tokenized as normal text, not as special tokens with separate embedding vectors. In models like LLaMA3 where the tokenizer adds tags as special tokens, `chat_format` is not needed, unless you want to structure messages in a particular way for inference. If the placeholder variable
```

Reply:
It is a bit long, but I think it's useful. Gonna mostly adopt but will leave out the bit about separate embedding vectors